app/forms/user_form.rb

```diff
@@ -0,0 +1,49 @@
+class UserForm < SimpleDelegator
+
+  include ActiveModel::Validations
+
+  def user
+    __getobj__
+  end
+
+  def valid?
+    success = [super, user.valid?].all?
+
+    user.errors.each do |k, error_messages|
+      errors.add(k, error_messages)
+    end
+
+    success
+  end
+
+  def save
+    set_password if user.new_record?
+
+    valid? && user.save
+  end
+
+  def update_attributes(params)
+    user.assign_attributes(params)
+    save
+  end
+
+  def admin_editable?
+    user.email_user?
+  end
+
+  def username_editable?
+    false
+  end
+
+  private
+
+  def set_password
+    user.password = generate_new_password
+  end
+
+  def generate_new_password
```

Comment:
I feel like this becomes a little awkward overriding this to return `nil` in DC, but I'm not sure I have a better solution. I was thinking [ActiveModel::Callbacks](http://api.rubyonrails.org/classes/ActiveModel/Callbacks.html), but that's probably overkill.